lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -7740,8 +7751,10 @@ PullbackEmitter::accumulateAdjointsDirect(AdjointValue lhs, AdjointValue rhs,
   // (x, y)
   case AdjointValueKind::Aggregate:
     switch (rhs.getKind()) {
-    // (x, y) + z => (x + z.0, y + z.1)
+    // (x, y) + z => (z.0 + x, z.1 + y)
     case AdjointValueKind::Concrete:
+      return accumulateConcreteAndAggregateAdjointsDirect(
```

Comment:
Have you considered, instead, recurse into the same function with swapped arguments?

Reply:
This sounds preferable to me!

Reply:
Done!